crates/red_knot_python_semantic/src/types.rs

```diff
@@ -6493,6 +6493,13 @@ struct OverloadedFunction<'db> {
     implementation: Option<FunctionType<'db>>,
 }
 
+impl<'db> OverloadedFunction<'db> {
+    /// Returns an iterator over all overloads and the implementation, in that order.
+    fn all(&self) -> impl Iterator<Item = &FunctionType<'db>> {
+        self.overloads.iter().chain(self.implementation.as_ref())
```

Comment:
I don't know if it makes much difference, but you could consider returning owned `FunctionType`s here, since they're just `u32`s under the hood

```suggestion
    fn all(&self) -> impl Iterator<Item = FunctionType<'db>> + '_ {
        self.overloads.iter().copied().chain(self.implementation)
```